### ISSUE TITLE
MockAwareDocTestFinder._find_lineno is stale

### ISSUE THREAD

**Issue by jaraco** [pytest-dev/pytest#12430]
In doctest, the comment for _find_lineno is stale:

https://github.com/pytest-dev/pytest/blob/043ff9abc657124db3504d3604e16ddebfe6e28f/src/_pytest/doctest.py#L514-L520

issue17446 resolves to python/cpython#61648, the fix which was released in Python 3.11. #8796 links to 0191563fd605a839e9e9c299605c52725b4da17a, which indicates that the method has two purposes, one to work around the aforementioned cpython issue and another "to be reported upstream," though I don't see that it was reported upstream. Does it still need to be reported upstream?

Doing a bit of testing, the doctests all pass on Python 3.11 and later with that method removed, so I'm inclined to say there's no outstanding work to be done except to suppress the use of this workaround after Python 3.11.

**Comment by The-Compiler** [comment]
At the sprint, we just ran into the selftests failing for @webknjaz, and it was because he happened to be running the testsuite with Python 3.11.8.

You said:

> the fix which was released in Python 3.11

but the fix actually only made it into Python [3.11.9](https://www.python.org/downloads/release/python-3119/) and [3.12.3](https://www.python.org/downloads/release/python-3124/), which are only around 2 months old now.

@jaraco I suppose it'd make sense to still use the workaround on 3.11 and 3.12 versions before those patch releases, no?

**Comment by jaraco** [comment]
Yes, I agree. I did not trace the origin more deeply than to look at minor versions. I think you're right and the workaround should be enabled for the specified patch versions or made dependent on Python 3.13.

**Comment by webknjaz** [comment]
Regression fix: https://github.com/pytest-dev/pytest/pull/12471

**Comment by webknjaz** [comment]
@The-Compiler could you assign me here?
